### PR TITLE
feat(haptics): add `rigid` and `soft` impact types

### PR DESCRIPTION
**Pull request by rodperottoni**
# Why

Since iOS 13 two new impact styles have been added: `soft` and `rigid`. I'd like to use these impact styles in my expo-powered app but currently can't without a patch.

# How

iOS: I simply added the new impact types to the existing Swift Module
Android: I don't know what an equivalent of `soft` and `rigid` impact types would feel like on Android, so I simply used the existing `light`/`medium` as the equivalent of `soft/rigid`.

# Test Plan

This can only be tested on a real device. 
Run the `native-component-list` app on a real device and go to the Haptics screen. Tap the "Soft" and "Rigid" buttons.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).
- [x] `native-component-list` project was updated with the new APIs

### CODE REVIEW

File: packages/expo-haptics/android/src/main/java/expo/modules/haptics/arguments/HapticsImpactType.kt [hunk 1]
```diff
@@ -7,11 +7,21 @@ object HapticsImpactType {
...
     "medium" to HapticsVibrationType(
       longArrayOf(0, 43),
       intArrayOf(0, 50),
       longArrayOf(0, 43)
     ),
+    "rigid" to HapticsVibrationType(
+      longArrayOf(0, 43),
+      intArrayOf(0, 50),
+      longArrayOf(0, 43)
+    ),
```
Comment (lukmccall) [inline packages/expo-haptics/android/src/main/java/expo/modules/haptics/arguments/HapticsImpactType.kt:24]: I'm not sure if we should tune values for the new options 🤔

**Review by lukmccall — COMMENTED** [review]
Looks good!
Thanks for your contribution. 
Can you add a changelog entry to `expo-haptics/CHANGELOG.md`?

File: packages/expo-haptics/src/Haptics.types.ts [hunk 2]
```diff
@@ -36,4 +36,12 @@ export enum ImpactFeedbackStyle {
    * A collision between large, heavy user interface elements.
    */
   Heavy = 'heavy',
+  /**
+   * A collision between user interface elements that are soft, exhibiting a large amount of compression or elasticity.
```
Comment (amandeepmittal) [inline packages/expo-haptics/src/Haptics.types.ts:40]:
```suggestion
   * A collision between user interface elements that are soft and show a large amount of compression or elasticity.
```

We should avoid using `,` for adding a pause to the sentence. It makes it unclear and a bit harder to understand. Either break it into two sentences or use a conjunction like the above suggestion.

File: packages/expo-haptics/src/Haptics.types.ts [hunk 3]
```diff
@@ -36,4 +36,12 @@ export enum ImpactFeedbackStyle {
...
+   */
+  Soft = 'soft',
+  /**
+   * A collision between user interface elements that are rigid, exhibiting a small amount of compression or elasticity.
```
Comment (amandeepmittal) [inline packages/expo-haptics/src/Haptics.types.ts:44]:
```suggestion
   * A collision between user interface elements that are rigid and show a small amount of compression or elasticity.
```

File: packages/expo-haptics/src/Haptics.types.ts (hunk 2 above)
Comment (rodperottoni) [reply packages/expo-haptics/src/Haptics.types.ts:40]: This doc was copied straight from the official Apple docs. Also, the usage of comma here implies the "large amount of elasticity" is describing the impact itself, whereas your suggestion makes it aound like "the component shows a large amount of elasticity".

Comment (amandeepmittal) [reply packages/expo-haptics/src/Haptics.types.ts:40]: Got it. Yeah, the changes I suggested do alter the context. I didn't have context and thought it was related to the impact of UI elements. I don't think it is trivial when it comes to communicating in a clear way as to what something does. However, the context you've provided now, it helps me understanding this and also recognize the same pattern as you mentioned in enum definitions.

Comment (rodperottoni) [reply packages/expo-haptics/src/Haptics.types.ts:40]: Thanks :D

**Review by lukmccall — APPROVED** [review]
LGTM ✅ 
I'll fix CI on the main.

**Pull request merged**